### PR TITLE
[SPARK-28704][SQL][TEST] Add back Skiped HiveExternalCatalogVersionsSuite in HiveSparkSubmitSuite at JDK9+

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
We skip test HiveExternalCatalogVersionsSuite when testing with JAVA_9 or later because our previous version does not support JAVA_9 or later. We now add it back since we have a version supports JAVA_9 or later.

### Why are the changes needed?

To recover test coverage.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Check CI logs.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Thank you for this efforts, @AngersZhuuuu !

**Comment by dongjoon-hyun** [comment]
Retest this please.

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogVersionsSuite.scala [hunk 1]
```diff
@@ -252,7 +244,9 @@ object PROCESS_TABLES extends QueryTest with SQLTestUtils {
       // do not throw exception during object initialization.
       case NonFatal(_) => Seq("3.0.1", "2.4.7") // A temporary fallback to use a specific version
     }
-    versions.filter(v => v.startsWith("3") || !TestUtils.isPythonVersionAtLeast38())
+    versions
+      .filter(v => v.startsWith("3") || !TestUtils.isPythonVersionAtLeast38())
+      .filter(v => v.startsWith("3") || !SystemUtils.isJavaVersionAtLeast(JavaVersion.JAVA_11))
```
Comment (dongjoon-hyun) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogVersionsSuite.scala:249]: nit. JavaVersion.JAVA_9 might be better.

**Comment by dongjoon-hyun** [comment]
BTW, @AngersZhuuuu . The title is correct?
```
Skip HiveExternalCatalogVersionsSuite and 3 tests in HiveSparkSubmitSuite at JDK9+
```

**Comment by dongjoon-hyun** [comment]
Could you take a look at `HiveExternalCatalogVersionsSuite` failure?

**Comment by AngersZhuuuu** [comment]
retest this please

**Comment by AngersZhuuuu** [comment]
> Could you take a look at `HiveExternalCatalogVersionsSuite` failure?

updated, seems latest UT failed not this issue?

**Comment by AngersZhuuuu** [comment]
retest this please

**Comment by AngersZhuuuu** [comment]
retest this please

**Comment by AngersZhuuuu** [comment]
ping @dongjoon-hyun All pass now.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you so much, @AngersZhuuuu .
Merged to master for Apache Spark 3.1.

**Comment by dongjoon-hyun** [comment]
Ur, it seems that GitHub Action failed with `java.io.FileNotFoundException: Source '/home/runner/.ivy2/jars/org.apache.zookeeper_zookeeper-3.4.6.jar'`
```
2020-11-20T21:46:53.3518362Z [0m[[0m[0minfo[0m] [0m[0m  2020-11-20 13:46:51.755 - stdout> : java.io.FileNotFoundException: Source '/home/runner/.ivy2/jars/org.apache.zookeeper_zookeeper-3.4.6.jar' does not exist[0m
2020-11-20T21:46:53.3520542Z [0m[[0m[0minfo[0m] [0m[0m  2020-11-20 13:46:51.755 - stdout> 	at org.apache.commons.io.FileUtils.copyFile(FileUtils.java:1074)[0m
2020-11-20T21:46:53.3522853Z [0m[[0m[0minfo[0m] [0m[0m  2020-11-20 13:46:51.755 - stdout> 	at org.apache.commons.io.FileUtils.copyFileToDirectory(FileUtils.java:1013)[0m
2020-11-20T21:46:53.3525435Z [0m[[0m[0minfo[0m] [0m[0m  2020-11-20 13:46:51.755 - stdout> 	at org.apache.commons.io.FileUtils.copyFileToDirectory(FileUtils.java:977)[0m
2020-11-20T21:46:53.3528409Z [0m[[0m[0minfo[0m] [0m[0m  2020-11-20 13:46:51.755 - stdout> 	at org.apache.spark.sql.hive.client.IsolatedClientLoader$.$anonfun$downloadVersion$4(IsolatedClientLoader.scala:139)[0m
2020-11-20T21:46:53.3531793Z [0m[[0m[0minfo[0m] [0m[0m  2020-11-20 13:46:51.755 - stdout> 	at 
```

**Comment by dongjoon-hyun** [comment]
And, it seems to happen in a clean installation setup like GitHub Action. I'll revert this quickly. Sorry, @AngersZhuuuu .

**Comment by AngersZhuuuu** [comment]
> Ur, it seems that GitHub Action failed with `java.io.FileNotFoundException: Source '/home/runner/.ivy2/jars/org.apache.zookeeper_zookeeper-3.4.6.jar'`
> 
> ```
> 2020-11-20T21:46:53.3518362Z �[0m[�[0m�[0minfo�[0m] �[0m�[0m  2020-11-20 13:46:51.755 - stdout> : java.io.FileNotFoundException: Source '/home/runner/.ivy2/jars/org.apache.zookeeper_zookeeper-3.4.6.jar' does not exist�[0m
> 2020-11-20T21:46:53.3520542Z �[0m[�[0m�[0minfo�[0m] �[0m�[0m  2020-11-20 13:46:51.755 - stdout> 	at org.apache.commons.io.FileUtils.copyFile(FileUtils.java:1074)�[0m
> 2020-11-20T21:46:53.3522853Z �[0m[�[0m�[0minfo�[0m] �[0m�[0m  2020-11-20 13:46:51.755 - stdout> 	at org.apache.commons.io.FileUtils.copyFileToDirectory(FileUtils.java:1013)�[0m
> 2020-11-20T21:46:53.3525435Z �[0m[�[0m�[0minfo�[0m] �[0m�[0m  2020-11-20 13:46:51.755 - stdout> 	at org.apache.commons.io.FileUtils.copyFileToDirectory(FileUtils.java:977)�[0m
> 2020-11-20T21:46:53.3528409Z �[0m[�[0m�[0minfo�[0m] �[0m�[0m  2020-11-20 13:46:51.755 - stdout> 	at org.apache.spark.sql.hive.client.IsolatedClientLoader$.$anonfun$downloadVersion$4(IsolatedClientLoader.scala:139)�[0m
> 2020-11-20T21:46:53.3531793Z �[0m[�[0m�[0minfo�[0m] �[0m�[0m  2020-11-20 13:46:51.755 - stdout> 	at 
> ```

image

In the build information it download zookeeper successfully. Ur a little strange, what should I need to do next?

**Comment by AngersZhuuuu** [comment]
> And, it seems to happen in a clean installation setup like GitHub Action. I'll revert this quickly. Sorry, @AngersZhuuuu .

It's ok,  do I need to open a new PR to test GitAction？

**Comment by dongjoon-hyun** [comment]
Yes. Please, @AngersZhuuuu . We can focus on the `GitHub Action` result only.

**Pull request closed without merging**